From 5202a01bcede4af80cf250a4e5cbed05128c8fa9 Mon Sep 17 00:00:00 2001
From: Sai Kumar Sanagavarapu <ssanagav@codeaurora.org>
Date: Mon, 21 Dec 2015 15:51:03 +0530
Subject: [PATCH 48/67] cameraservice: Change the error mapping for
 ERROR_CAMERA_DEVICE.

In case of API1-HAL3, if HAL reports CAMERA3_MSG_ERROR_DEVICE, it
is getting mapped to CAMERA_ERROR_UNKNOWN in the service layer.
But this particular error code is not handled properly even in
applications like GoogleCamera (Android default camera app).

For example if camera deamon gets killed when GoogleCamera is
previewing, since its not handling the error properly, preview
freeze is seen for long time followed by mediaserver crash if
any user interaction happens meanwhile. Ideally, app should
close gracefully based on the error received.

So, map CAMERA3_MSG_ERROR_DEVICE to CAMERA_ERROR_SERVER_DIED,
so that app and mediaserver gets closed gracefully.
---
 services/camera/libcameraservice/api1/Camera2Client.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/services/camera/libcameraservice/api1/Camera2Client.cpp b/services/camera/libcameraservice/api1/Camera2Client.cpp
index 2095e0c4e..0e62962a6 100644
--- a/services/camera/libcameraservice/api1/Camera2Client.cpp
+++ b/services/camera/libcameraservice/api1/Camera2Client.cpp
@@ -1769,8 +1769,6 @@ void Camera2Client::notifyError(int32_t errorCode,
             err = CAMERA_ERROR_RELEASED;
             break;
         case hardware::camera2::ICameraDeviceCallbacks::ERROR_CAMERA_DEVICE:
-            err = CAMERA_ERROR_UNKNOWN;
-            break;
         case hardware::camera2::ICameraDeviceCallbacks::ERROR_CAMERA_SERVICE:
             err = CAMERA_ERROR_SERVER_DIED;
             break;
-- 
2.11.0

